### ISSUE TITLE
MCServer Crash

### ISSUE THREAD

**Issue by falkseidl** [cuberite/cuberite#1970]
At one of my MCServer servers encountered often a crash, now I founded out something about it:
```
[22:58:50] Growing melon/pumpkin overwriting air, growing on grass
[22:59:37] InflateString: inflation failed: -3 ("incorrect data check").
Warn [22:59:37] Uncompressing chunk [9, 1] failed: -3
Err  [22:59:37] LoadChunkFromData (285): Loading chunk [9, 1] from file r.0.0.mca failed. The server will now abort in order to avoid further data loss. Please add the reported file and this message to the issue report.
Err  [22:59:37]   D:    | MCServer has encountered an error and needs to close
Err  [22:59:37] Details | SIGSEGV: Segmentation fault
Err  [22:59:37]   D:    | MCServer has encountered an error and needs to close
Err  [22:59:37] Details | SIGABRT: Server self-terminated due to an internal fault
[22:59:37] Shutting down server...
```
Crashes before this one, were triggered by destroying a pumpkin block, but not every pumpkin triggered a crash, only a special one (didn't know why). In creative and also survival.
Removing per worldedit had helped.

But now it crashes again, didn't know if its again a pumkin or so, but the log sounds like this.

**Comment by worktycho** [comment]
Can you upload the mca file and link to it?

**Comment by falkseidl** [comment]
Yes, of course, but they are have a really big filesize.
https://minefight.org/ftp/?dir=region

**Comment by falkseidl** [comment]
Lol, for some reason the webpage is not listing the files.
I've created a tar file:
http://minefight.org/region.tar.gz

**Comment by falkseidl** [comment]
I now have destroyed a pumkin, and the server crashed again. Strange. No error in log.

**Comment by falkseidl** [comment]
Removing all pumkin plants+pumpkins works, no more crashes, also with new ones at the same place.

**Comment by falkseidl** [comment]
After a restart of the server, the same issue came back:
https://youtu.be/4kpg-SdGlz8
https://youtu.be/b4-vpqpXMIQ

I also created a video where I try to replace the pumpkins with stone, and it's very interesting to see what happens:
https://youtu.be/Fr6sqNl-qLI

**Comment by madmaxoft** [comment]
The `//replace` command apparently doesn't work because what you're actually asking it to do is "replace all pumpkins that have a meta value of 0 with stone". So only those facing one specific direction are replaced.

**Comment by SafwatHalaby** [comment]
Just a reminder to the new flow:  when this is resolved, it should be merged to testing.

**Comment by falkseidl** [comment]
Are there now branches? If yes, how to compile builds from the different branches?

**Comment by worktycho** [comment]
There are branches. To compile master just keep what your doing. To compile testing you need to checkout the testing branch before compiling.

**Comment by falkseidl** [comment]
Thanks!

**Comment by SafwatHalaby** [comment]
It's all "under construction", you can see [the conversation here](http://forum.mc-server.org/showthread.php?tid=1847). We already have "master" and "testing", once testing is stable for a while it is merged into "stable", there isn't a stable yet because this is all brand new and no "testing" has been graduated to "Stable" yet.

Right now, Master and Testing are pretty much identical stability-wise :P
I suggest you just keep doing what  you're doing until this business is ready.

**Comment by SafwatHalaby** [comment]
What's the status of this?

**Comment by falkseidl** [comment]
Pumkin/Melon growing, planting and destroying works without crash.

**Comment by SafwatHalaby** [comment]
Is this just pure luck or can we safely close this and get 1 step closer to the stable release?

**Comment by falkseidl** [comment]
Some days ago I asked the players to try melons and pumpkins and no crash happened, they also sayed that it is working fine. So I assume, that it is now fully fixed.

**Comment by SafwatHalaby** [comment]
@tigerw @worktycho @madmaxoft Did anyone actually fix this or is @falkseidl lucky? Can we close it?

**Comment by madmaxoft** [comment]
I'll write the code for offloading mangled chunks into a separate file and regenerating. I don't think there's anything else left in this issue.

**Comment by madmaxoft** [comment]
Chunk offloading finished in PR #2400
